### PR TITLE
[flash_ctrl] Documentation update

### PR DESCRIPTION
**Pull request by tjaychen**
- clarify behavior and usage of flash read buffers

Signed-off-by: Timothy Chen <timothytim@google.com>

### CODE REVIEW

File: hw/ip/flash_ctrl/doc/_index.md [hunk 1]
```diff
@@ -405,6 +405,19 @@ The following diagram shows how the flash read pipeline timing works.
...
 The third host requests immediately hits in the local cache and responds in order after the first two.
 
+### Flash Buffer
+
+The flash buffer is a small read-only memory that holds recently read flash words.
+This is needed when the flash word is wider than a bus word.
+The flash access time is amortized across the the entire flash word if software accesses in a mostly
+linear sequence.
+
+The flash buffer has a round robin replacement policy when more flash words are read.
```
Comment (eunchan) [inline hw/ip/flash_ctrl/doc/_index.md:415]: So, the flash buffer size exceeds the flash word size?

**Review by eunchan — APPROVED** [review]

File: hw/ip/flash_ctrl/doc/_index.md (hunk 1 above)
Comment (tjaychen) [reply hw/ip/flash_ctrl/doc/_index.md:415]: o no i should clarify this.
It's more flash word size > bus word size.
And there are multiple flash words that make up the flash buffer.

Comment (tjaychen) [reply hw/ip/flash_ctrl/doc/_index.md:415]: can you see if the latest explains better @eunchan ?

**Comment by eunchan** [comment]
LGTM :)

File: hw/ip/flash_ctrl/doc/_index.md [hunk 2]
```diff
@@ -405,6 +405,28 @@ The following diagram shows how the flash read pipeline timing works.
...
 
+### Flash Buffer
+
+The flash buffer is a small read-only memory that holds multiple entries of recently read flash words.
+This is needed when the flash word is wider than a bus word.
+The flash access time is amortized across the the entire flash word if software accesses in a mostly
+linear sequence.
+
+The flash buffer has a round robin replacement policy when more flash words are read.
+When an erase / program is issued to the flash, the entries are evicted to ensure new words are fetched.
```
Comment (msfschaffner) [inline hw/ip/flash_ctrl/doc/_index.md:416]: This is an unconditional eviction, i.e., not based on which address we erase, program, right?

**Review by msfschaffner — APPROVED** [review]
LGTM

File: hw/ip/flash_ctrl/doc/_index.md (hunk 2 above)
Comment (tjaychen) [reply hw/ip/flash_ctrl/doc/_index.md:416]: let me clarify that, it is conditional.

**Review by nikola-miladinovic — APPROVED** [review]

**Pull request merged**